samcli/lib/utils/sam_logging.py

```diff
@@ -28,9 +35,12 @@ def configure_logger(logger, formatter, level):
         if handlers:
             log_stream_handler = handlers[0]
         else:
-            log_stream_handler = logging.StreamHandler()
+            log_stream_handler = (
+                RichHandler(console=Console(stderr=True), show_time=False, show_path=False, show_level=False)
+                if sys.stderr.isatty() and os.getenv(LOGGING_COLOR_ENV_VAR) != "1"
```

Comment:
Shouldn't this be an `or`?

Reply:
It has be "and". You could be on an interactive terminal and have NO_COLOR set to 1, if that is the case you get the normal stream logger initialized.